core/src/main/scala/kafka/server/ClientMetricsManager.scala

```diff
@@ -0,0 +1,93 @@
+package kafka.server
+
+import kafka.Kafka.info
+import kafka.metrics.clientmetrics.{ClientMetricsCache, ClientMetricsConfig, CmClientInformation, CmClientInstanceState}
+import kafka.network.RequestChannel
+import kafka.server.ClientMetricsManager.{CM_CACHE_MAX_SIZE, getSupportedCompressionTypes}
+import org.apache.kafka.common.Uuid
+import org.apache.kafka.common.message.GetTelemetrySubscriptionsResponseData
+import org.apache.kafka.common.protocol.Errors
+import org.apache.kafka.common.record.CompressionType
+import org.apache.kafka.common.requests.{GetTelemetrySubscriptionRequest, GetTelemetrySubscriptionResponse}
+
+import java.util.Properties
+import scala.collection.mutable.ListBuffer
+import scala.jdk.CollectionConverters._
+
+object ClientMetricsManager {
+  val CM_CACHE_GC_INTERVAL = 5 * 60 * 1000 // 5 minutes
+  val CM_CACHE_MAX_SIZE = 1024
```

Comment:
This limit seems low, a single broker can have more than 1024 client connections, right?

Reply:
What do you suggest for cache size?